### PR TITLE
genesys: scaler: Get real software version

### PR DESCRIPTION
**Pull request by gportay**
The plugin gets the panel-type (i.e. hardware revision) as the firmware
version.

This gets the real firmware version, and appends the panel-type to the
instance-id string as the blobs are specific to panel-types.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by gportay** [comment]
@adamgene, could you confirm and test this is correct for the HP monitors?

File: plugins/genesys/fu-genesys-scaler-device.c [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021 Gaël PORTAY <gael.portay@collabora.com>
+ * Copyright (C) 2021-2022 Gaël PORTAY <gael.portay@collabora.com>
```
Comment (hughsie) [inline plugins/genesys/fu-genesys-scaler-device.c:2]: in general we only list the start year in source files

File: plugins/genesys/fu-genesys-scaler-device.c [hunk 2]
```diff
@@ -1417,17 +1551,28 @@ fu_genesys_scaler_device_probe(FuDevice *device, GError **error)
...
 	fu_device_set_version(device, version);
 	fu_device_set_version_format(device, FWUPD_VERSION_FORMAT_PLAIN);
 	fu_device_set_logical_id(device, "scaler");
 
 	/* add instance ID */
 	fu_device_add_instance_str(device, "MSTAR", "TSUM_G");
 	fu_device_add_instance_strup(device, "PUBKEY", guid);
-	fu_device_build_instance_id(device, NULL, "GENESYS_SCALER", "MSTAR", "PUBKEY", NULL);
+	fu_device_add_instance_strup(device, "PANELTYPE", panel_type);
+	fu_device_build_instance_id(device,
```
Comment (hughsie) [inline plugins/genesys/fu-genesys-scaler-device.c:1569]: I think we probably want *both* instance IDs, no?

File: plugins/genesys/fu-genesys-scaler-device.c [hunk 3]
```diff
@@ -1417,17 +1551,28 @@ fu_genesys_scaler_device_probe(FuDevice *device, GError **error)
...
+	version = g_strdup_printf("%d.%d.%d.%d", ver.stage, ver.model, ver.major, ver.minor);
 	fu_device_set_version(device, version);
 	fu_device_set_version_format(device, FWUPD_VERSION_FORMAT_PLAIN);
 	fu_device_set_logical_id(device, "scaler");
 
 	/* add instance ID */
 	fu_device_add_instance_str(device, "MSTAR", "TSUM_G");
 	fu_device_add_instance_strup(device, "PUBKEY", guid);
-	fu_device_build_instance_id(device, NULL, "GENESYS_SCALER", "MSTAR", "PUBKEY", NULL);
+	fu_device_add_instance_strup(device, "PANELTYPE", panel_type);
```
Comment (hughsie) [inline plugins/genesys/fu-genesys-scaler-device.c:1568]: maybe `PANEL`?

File: plugins/genesys/fu-genesys-scaler-device.c (hunk 2 above)
Comment (gportay) [reply plugins/genesys/fu-genesys-scaler-device.c:1569]: Indeed, it makes sens to have both.

File: plugins/genesys/fu-genesys-scaler-device.c (hunk 1 above)
Comment (gportay) [reply plugins/genesys/fu-genesys-scaler-device.c:2]: done.

File: plugins/genesys/fu-genesys-scaler-device.c (hunk 3 above)
Comment (gportay) [reply plugins/genesys/fu-genesys-scaler-device.c:1568]: This field refers to the attribute of `panel_type` in structure `FuGenesysMtkRsaHeader`; I like they have the same name if that possible. However, I like `PANEL`.

I would prefer `REV` or `HWREV`; but it is not clear to me if the that attribute always refers to a panel or not :/.

@adamgene, any opinion?

Comment (adamgene) [reply plugins/genesys/fu-genesys-scaler-device.c:1568]: >I would prefer `REV` or `HWREV`; but it is not clear to me if the that attribute always refers to a panel or not :/.
>

I researched the document for the `FuGenesysMtkRsaHeader`, it says `HW FW version`. But I found the string in HP definition is `REV`. The detail format can ask HP.

IMO, It is means panel revision for MSTAR scaler chip.

Comment (hughsie) [reply plugins/genesys/fu-genesys-scaler-device.c:1568]: REV won't work as we use that for the USB revision already.

**Review by hughsie — DISMISSED** [review]

**Comment by hughsie** [comment]
@gportay ohh i see the CI failure now:

      1501 |                                     "error dddci length too large, got 0x%lx, expected <= 0x%zx

just use 0x%x and then cast the parameter to (guint) manually.

File: plugins/genesys/fu-genesys-scaler-device.c (hunk 3 above)
Comment (adamgene) [reply plugins/genesys/fu-genesys-scaler-device.c:1568]: How about PANEL_REV?

Comment (hughsie) [reply plugins/genesys/fu-genesys-scaler-device.c:1568]: I like PANELREV more, but that works for me.

Comment (adamgene) [reply plugins/genesys/fu-genesys-scaler-device.c:1568]: `PANELREV` is works for me too.

Comment (gportay) [reply plugins/genesys/fu-genesys-scaler-device.c:1568]: so let's go for `PANELREV`.

**Comment by gportay** [comment]
> just use 0x%x and then cast the parameter to (guint) manually.

done

**Pull request merged**